### PR TITLE
Update napalm.py

### PR DESCRIPTION
**Pull request by aesposito91**
### What does this PR do?
Address issue - https://github.com/saltstack/salt/issues/47203

This will allow napalm proxy grains to update with a saltutil.refresh_grains

### What issues does this PR fix or reference?
This will allow napalm proxy grains to update with a saltutil.refresh_grains

### Previous Behavior
Grains would not update.

### New Behavior
Grains will update.

### Tests written?

No

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by mirceaulinic** [comment]
This bugfix should target the 2017.7 branch and ported into 2018.3 then (or vice-versa).

File: salt/proxy/napalm.py [hunk 1]
```diff
@@ -245,7 +245,7 @@ def get_grains():
         facts = call('get_facts', **{})
         DETAILS['grains_cache'] = facts
 
-    return DETAILS.get('grains_cache', {})
+    return call('get_facts', **{})
```
Comment (mirceaulinic) [inline salt/proxy/napalm.py]: This function should only return this (remove the rest of the code).

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
Please make sure you keep only that return, and change the branch to 2017.7 or 2018.3.

**Review by gtmanfred — CHANGES_REQUESTED** [review]
This needs to be rebased to remove the extra commits that are not related to what you are changing.

**Comment by aesposito91** [comment]
Will resubmit.

**Pull request closed without merging**